go/vt/vtgateproxy/discovery.go

```diff
@@ -96,60 +91,87 @@ func RegisterJsonDiscovery() {
 	fmt.Printf("Registered %v scheme\n", jsonDiscovery.Scheme())
 }
 
-type resolveFilters struct {
-	gate_type string
-	az_id     string
-}
+type hostFilters = map[string]string
 
 // exampleResolver is a
 // Resolver(https://godoc.org/google.golang.org/grpc/resolver#Resolver).
-type resolveJSONGateConfig struct {
-	target          resolver.Target
-	cc              resolver.ClientConn
-	jsonPath        string
-	ticker          *time.Ticker
-	rand            *rand.Rand // safe for concurrent use.
-	num_connections int
-	filters         resolveFilters
+type JSONGateConfigResolver struct {
+	target   resolver.Target
+	cc       resolver.ClientConn
+	jsonPath string
+	ticker   *time.Ticker
+	rand     *rand.Rand // safe for concurrent use.
+	filters  hostFilters
 }
 
-type discoverySlackAZ struct{}
-type discoverySlackType struct{}
+type matchesFilter struct{}
 
-func (r *resolveJSONGateConfig) loadConfig() (*[]resolver.Address, []byte, error) {
-	config := []DiscoveryHost{}
-	fmt.Printf("Loading config %v\n", r.jsonPath)
+func (r *JSONGateConfigResolver) loadConfig() (*[]resolver.Address, []byte, error) {
+	pairs := []map[string]interface{}{}
+	fmt.Printf("Loading config %v config for %v connections\n", r.jsonPath, *numConnectionsInt)
 
 	data, err := os.ReadFile(r.jsonPath)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	err = json.Unmarshal(data, &config)
+	err = json.Unmarshal(data, &pairs)
```

Comment:
Another naming nit -- I think `pairs` is misleading. Can we change to `hostInfo` or something like that?